src/CMakeLists.txt

```diff
@@ -516,6 +516,7 @@ ENDIF()
 # Main executable/library setup
 IF(GENERATE_STELMAINLIB)
      ADD_LIBRARY(stelMain SHARED ${stellarium_lib_SRCS} ${stellarium_RES_CXX})
+     target_precompile_headers(stelMain PUBLIC core/VecMath.hpp core/StelUtils.h external/SpoutLibrary.h core/*.hpp core/planetsephems/*.h core/planetsephems/*.hpp core/modules/*.hpp gui/*.hpp)
```

Comment:
This code will cause problem when ENABLE_SPOUT==Off

Reply:
This was a detail in the first commit. Ignore. Please see current state. We should have a few Qt classes included, and maybe VecMath.hpp, StelUtils.hpp and StelCore.hpp, but the critical issue seems to be to remove all C files from the stelMain target as (precompiled) c++ headers must not be included in them. Before that, building fails.